stl/inc/source_location

```diff
@@ -20,15 +20,25 @@ _STL_DISABLE_CLANG_WARNINGS
 #pragma push_macro("new")
 #undef new
 
+#ifndef _USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION
+#ifdef __EDG__ // TRANSITION, DevCom-10199227
+#define _USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION 0
+#elif defined(__clang__) // TRANSITION, Clang 17 has this builtin
+#define _USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION __has_builtin(__builtin_FUNCSIG)
+#else // ^^^ Clang / MSVC vvv
+#define _USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION 1
+#endif // ^^^ MSVC ^^^
+#endif // ^^^ !defined(_USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION) ^^^
+
 _STD_BEGIN
 _EXPORT_STD struct source_location {
     _NODISCARD static consteval source_location current(const uint_least32_t _Line_ = __builtin_LINE(),
         const uint_least32_t _Column_ = __builtin_COLUMN(), const char* const _File_ = __builtin_FILE(),
-#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10199227 and LLVM-58951
-        const char* const _Function_ = __builtin_FUNCTION()
-#else // ^^^ workaround / no workaround vvv
+#if _USE_DETAILED_FUNCTION_NAME_IN_SOURCE_LOCATION
         const char* const _Function_ = __builtin_FUNCSIG()
-#endif // TRANSITION, DevCom-10199227 and LLVM-58951
+#else

```

Comment:
Why did this `#else` and `#endif` need comments before, but not now?

Reply:
`workaround / no workaround` is a nice reminder when we plan to remove one of the branches in the future. Now, both branches are permanent, each of them is 1 line with no surrounding preprocessor logic, and "use detailed function name" is clearly readable so there's no real chance for confusion about which side is doing what.

I'd like to keep this as-is, using our usual convention that we can skip comments in this 1-line scenario.

Reply:
Great! I didn't think through the fact that this was a workaround but is no more.

Reply:
The tests have enough dense preprocessor logic that I'm finding I want "detailed / basic" in them - I think I'll do that in the product code for consistency. So you'll get comments after all :joy_cat: